src/main/java/frc/robot/subsystems/Vision.java

```diff
@@ -0,0 +1,206 @@
+
+        endPoints[0] = rectangle.tl();
+        endPoints[1] = rectangle.br();
+
+        height = rectangle.height;
+        width = rectangle.width;
+
+        centerPoints[0] = endPoints[1].x - (width / 2);
+        centerPoints[1] = endPoints[1].y - (height / 2);
+
+        targetCenter.set(centerPoints); // Set rectangle center
+
+        centerPoints[0] = image.width() / 2;
+        centerPoints[1] = image.height() / 2;
+
+        reticle.set(centerPoints); // Set reticle point
+
+        return endPoints;
+    }
+
+    private void drawMisalignedMasking(Mat image, Point [] endPoints, List<MatOfPoint> contours)
+    {
+        Imgproc.rectangle(image, endPoints[0], endPoints[1], Constants.RED);
+        Imgproc.circle(image, reticle, 3, Constants.RED, -1);
+        Imgproc.drawContours(image, contours, -1, Constants.WHITE, -1);
+    }
+
+    private void drawAlignedMasking(Mat image, Point [] endPoints, List<MatOfPoint> contours)
+    {
+        Imgproc.rectangle(image, endPoints[0], endPoints[1], Constants.GREEN);
+        Imgproc.circle(image, reticle, 3, Constants.GREEN, -1);
+        Imgproc.drawContours(image, contours, -1, Constants.GREEN, -1);
+    }
+
+    private void aqquireTarget(Mat image, Point [] endPoints, List<MatOfPoint> contours)    {
+
+        targetDistance = ((reticle.y - targetCenter.y) * -Constants.PIXEL_TO_METRES_RATIO) + Constants.TARGET_CENTER_DISTANCE; // Negative to counteract Mat pixel ordering (top to bottom), 162cm is when reticle is centered with targetCenter, 2 represents 2cm per pixel
+        calculatedShooterSpeed = (0.1633 * Math.pow(targetDistance, 3)) - (0.5561 * Math.pow(targetDistance, 2)) + (0.6154 * targetDistance) + 0.4097;
+
+
+        if ((reticle.x - Constants.AIM_BOT_ACCURACY) < targetCenter.x && targetCenter.x < (reticle.x + Constants.AIM_BOT_ACCURACY))
+        {
+            horizontallyAligned = true;
+            turnRight = false;
+            turnLeft = false;
+            
+            drawAlignedMasking(image, endPoints, contours);
+        }
+        else if ((reticle.x - Constants.AIM_BOT_ACCURACY) < targetCenter.x)
+        {
+            horizontallyAligned = false;
+            turnRight = true;
+            turnLeft = false;
+
+            drawMisalignedMasking(image, endPoints, contours);
+        }
+        else if (targetCenter.x < (reticle.x + Constants.AIM_BOT_ACCURACY))
+        {
+            horizontallyAligned = false;
+            turnLeft = true;
+            turnRight = false;
+
+            drawMisalignedMasking(image, endPoints, contours);
+        }
+    }
+
+    public void start()
+    {
+        ringlight.set(true);
+        runFeed = true;
+    }
+
+    public void stop()
+    {
+        ringlight.set(false);
+        runFeed = false;
+    }
+
+    public boolean getVisionEnabled()
+    {
+        return runFeed;
+    }
+
+    public boolean getHorizontallyAligned()
+    {
+        return horizontallyAligned;
+    }
```

Comment:
If there is no set method to go with this then just call is `isHorizontallyAligned()`